spec.md

```diff
@@ -0,0 +1,1597 @@
+* cannot redefine `(relation) A relates I[]`
+* `(relation) A relates I[] as J[]` redefines $`I <_! J`$, ***requiring*** that either $`I <_! J' \neq J`$ or $`I`$ has no direct super-role
+
+**Case ATT**
+* cannot redefine `attribute A`
+* `(attribute) A value V` redefines $`A < V`$
+* cannot redefine `(attribute) A sub B`
+
+**Case PLAYS**
+* cannot redefine `(type) A plays B:I`
+
+**Case OWNS**
+* cannot redefine `(type) A owns B`
+* cannot redefine `(type) A owns B[]`
+
+### Constraints
+
+_In each case, `redefine` redefines the postulated condition._
+
+**Case CARD**
+* `A relates I @card(n..m)`
+* `A plays B:I @card(n..m)`
+* `A owns B @card(n...m)`
+
+**Case CARD_LIST**
+* `A relates I[] @card(n..m)`
+* `A owns B[] @card(n...m)`
+
+**Case PLAYS_AS**
+* `A plays B as C`
+
+**Case OWNS_AS**
+* `A owns B as C`
+
+**Case UNIQUE**
+* cannot redefine `A owns B @unique`
+
+**Case KEY**
+* cannot redefine `A owns B @key`
+
+**Case SUBKEY**
+* cannot redefine `A owns B @subkey(<LABEL>)`
+
+**Case ABSTRACT**
+* cannot redefine `(type) B @abstract` 
+* cannot redefine `A plays B:I @abstract`
+* cannot redefine `A owns B @abstract` 
+* cannot redefine `A owns B[] @abstract` 
+* cannot redefine `B relates I @abstract`
+* cannot redefine `B relates I[] @abstract`
+
+**Case VALUES**
+* `A owns B @values(v1, v2)` 
+* `A owns B @regex(<EXPR>)` 
+* `A owns B @range(v1..v2)`
+* `A value B @values(v1, v2)` 
+* `A value B @regex(<EXPR>)` 
+* `A value B @range(v1..v2)`
+
+**Case DISTINCT**
+* cannot redefine `A owns B[] @distinct`
+* cannot redefine `B relates I[] @distinct`
+
+### Triggers
+
+_In each case, `redefine` redefines the triggered action._
+
+**Case DEP_DEL (CASCADE/INDEPEDENT)**
+* cannot redefine `(relation) B relates I @cascade`
+* cannot redefine `(relation) B @cascade`
+* cannot redefine `(attribute) B @independent`
+
+### Value types
+
+**Case PRIMITIVES**
+
+cannot redefine primitives
+
+**Case STRUCT**
+
+cannot redefine structs
+
+### Functions defs
+
+**Case STREAM_RET_FUN**
+
+cannot redefine stream-return functions.
+
+**Case SINGLE_RET_FUN**
+
+cannot redefine single-return functions.
+
+# Data instance languages
+
+This section first describes the satisfication semantics of match queries, obtained by substituting _variables_ in _patterns_ by concepts (_answers_) such that these patterns are _satisfied_. It is then described how instance in ERA types can be declared and further manipulated. Finally, the section describes the semantics of functions (the novelty over match semantics is the ability to declared functions recursively).
+
+## Pattern semantics
+
+### Basics: Variables, concept maps, satisfaction
+
+**Variables**
+
+* _Syntax_: vars start with `$`
+  * _Examples_: `$x`, `$y`, `$person`
+* _Usage_: vars appear as part of 
+  * statements: syntactic units of TypeQL (see Glossary)
+  * patterns: collection of statements, combined with logical connectives:
+    * `;` "and" (default), 
```

Comment:
This is not strictly true, i.e. `;` does not always map onto a conjunction operator. Understanding patterns requires 1) a thorough definition of statements, including simple vs composite, 2) the notion that all statements in a pattern are combined conjunctively, and 3) the notion that a disjunction, negation, or optional block can be resolved to a statement of some kind and so combine conjunctively with others. I don't think the simplification you've made here is justified.

Reply:
it's true in a technical sense. inductively set: 
* the empty pattern represents "true" ($\bot$), 
* `p; PATT` represents $`p \wedge PATT`$

and then add further operators:
* `not { PATT }` represents $`\neg (PATT)`$
* ...

Each statement represents a proposition `p`. The exact denotation is a different question :P